### ISSUE TITLE
unidiomatic-typecheck / C0123: false positive on `type() == type()`

### ISSUE THREAD

**Issue by DavidS-cloud** [PyCQA/pylint#7336]
### Bug description

# pylint: disable=missing-module-docstring,expression-not-assigned,undefined-variable
type(foo) == type(bar)

### Configuration

```ini
# happens with default config
```

### Command used

```shell
pylint type.py
```

### Pylint output

```shell
************* Module type
type.py:2:0: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck)

-----------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: -170.00/10, +170.00)
```

### Expected behavior

Either do not raise an issue on this code, or document in the check's description how to phrase this type test better.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.6 (main, Aug 10 2022, 11:19:32) [GCC 12.1.0]
```

**Comment by DanielNoord** [comment]
Please see https://pylint.pycqa.org/en/latest/user_guide/messages/convention/unidiomatic-typecheck.html

You could rewrite this as `isinstance(foo, type(bar))`.
